### ISSUE TITLE
Screen update sequence for 8a1234567890 is strangely long

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#72]
capture xvi's output with ```script -f``` and say
```
8a1234567890[Esc]
```
The screen output (on xterm with Debian's ncurses terminfo) is (broken into lines for legibility):
```
^[[?1049h^[[?12;25h^[[?1h^[=^[[H^[[0m^[[H^[[H^[[2J^[[H^[[0m^[[7m^[[125r
^[[25;1HNo File                                                                        ^[[0m
^[[H^J~^J^M~^J^M~^J[...]^J^M~^J^M~
^[[H1^M12^M123^M1234^M12345^M123456^M1234567^M12345678
^M123456789^M1234567890^M12345678901^M123456789012
^M1234567890123^M12345678901234^M123456789012345
^M12345678901234567890123456789012345678901234567890123456789012345678901234567890
```
This bug is about is the last line going
```
^M123456789012^M1234567890123^M12345678901234^M123456789012345
```
before repainting the whole line.

First, it doesn't need to repaint the whole line as the ten digits are typed (just printing the new character would do), then it goes potty when you hit [Esc], repainting the whole line for the first fifteen new characters and only then repainting all 80.

There are two bugs here:
* the repainting of the whole line instead of adding one char, which on a slow serial terminal would be very frustrating, if pretty, and
* it using a different code path for the first 15 characters (or the first 5 automatic characters) than for the last 65.

Note that eliminating the first bug will hide the second, which is not fatal but worrying.

**Comment by martinwguy** [comment]
See the tests at the end of https://github.com/martinwguy/xvi/issues/71#issuecomment-280633193 for another sub-optimal screen update sequence

**Comment by martinwguy** [comment]
The figure "15" comes from the calculation of cost_goto as strlen(CM)-1, which is 15 using xterm, whereas really it is from 6 to 8, depending on the coordinates.  That makes outputting the line seem more attractive than a cursor motion. 
Worse, xvi only uses "nd" capability (move forwards) if it is one character long, while under xterm it's "ESC[C", so is never used (and in fact repainting the existing character *is* shorter.)

What remains to be understood is why it repaints the whole line so may times when it is already correct as far as the cursor.

**Comment by martinwguy** [comment]
Fixed by 375dc48b, as this was caused by the useless v_advise() function.
